main.py

```diff
@@ -133,14 +134,58 @@ def discussion():
         except KeyError:
             return index(message=u"דיון לא חוקי")
     else:
-        marker = DiscussionMarker.parse(request.get_json(force=True))
-        db_session.add(marker)
+        obj = parse_json_request(DiscussionMarker, request)
+        return make_response(post_handler(obj))
+
+@app.route("/highlightpoints", methods=['POST'])
+@user_optional
+def highlightpoint():
+    highlight = parse_json_request(HighlightPoint, request)
+    if highlight is None:
+        make_response("")
+    # if it's a user gps type (user location), only handle a single post request per session
+    if int(highlight.type) == HighlightPoint.HIGHLIGHT_TYPE_USER_GPS:
+        if not SESSION_HIGHLIGHTPOINT_KEY in session:
+            session[SESSION_HIGHLIGHTPOINT_KEY] = "saved"
+        else:
+            return make_response("")
+
+    return make_response(post_handler(highlight))
+
+# Post handler for REST API
+def post_handler(obj):
+    try:
+        db_session.add(obj)
         db_session.commit()
-        return make_response(json.dumps(marker.serialize()))
+        return jsonify(obj.serialize())
+    except Exception as e:
+        logging.debug("could not handle a post for object:{0}, error:{1}".format(obj, e.message))
+        return ""
+
+# cls: ORM Model, implement parse and serialize functions.
+def parse_json_request(cls, request):
+    try:
+        data = get_json(request)
```

Comment:
If `get_json` had an exception, you'll get `None` here. Better just raise the exception further or check for `None`